pkg/tsdb/v3iotsdb.go

```diff
@@ -172,20 +172,21 @@ func (a *V3ioAdapter) Querier(_ context.Context, mint, maxt int64) (*querier.V3i
 	return querier.NewV3ioQuerier(a.container, a.logger, mint, maxt, a.cfg, a.partitionMngr), nil
 }
 
-func (a *V3ioAdapter) DeleteDB(configExists bool, force bool) error {
+func (a *V3ioAdapter) DeleteDB(configExists bool, force bool, fromTime int64, toTime int64) error {
 
-	path := a.partitionMngr.GetHead().GetTablePath()
-	a.logger.Info("Delete partition %s", path)
-	err := utils.DeleteTable(a.container, path, "", a.cfg.QryWorkers)
-	if err != nil && !force {
-		return err
+	partitions := a.partitionMngr.PartsForRange(fromTime, toTime)
+	for _, part := range partitions {
```

Comment:
@dinal i assume we dont need to delete partitions where toTime hits the middle of the partition, so maybe check if the partition.maxTime is < toTime